### PR TITLE
service: Split up session_management.py

### PR DESCRIPTION
**Pull request by bkeryan**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/measurement-services-python/blob/main/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Convert `session_management.py` into a sub-package and split it up into `_client.py`, `_constants.py`, `_reservation.py`, and `_types.py`.

Change gRPC conversion functions into private methods: `_from_grpc(_v1)` (class method) and `_to_grpc(_v1)` (instance method).

### Why should this Pull Request be merged?

`session_management.py` is getting too big.

Resolve circular import between `session_management.py` and driver-specific code.

### What testing has been done?

Ran ni-python-styleguide, mypy, and all tests.

### CODE REVIEW

File: ni_measurementlink_service/session_management/_client.py [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+)
+from ni_measurementlink_service.session_management._types import (
+    PinMapContext,
+    SessionInformation,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+class SessionManagementClient(object):
```
Comment (pbirkhol-ni) [inline ni_measurementlink_service/session_management/_client.py:33]: This class is going to be instantiated directly by customers in a code module, but the module name is prefaced with an underscore. If I were a client of this class I would hesitate to use it.

**Review by pbirkhol-ni — APPROVED** [review]

File: ni_measurementlink_service/session_management/_client.py (hunk 1 above)
Comment (bkeryan) [reply ni_measurementlink_service/session_management/_client.py:33]: If we make the submodules public, then we have to maintain compatibility with the submodules, and we introduce multiple ways to refer to these classes.

These classes are re-exported from `__init__.py` using `__all__`. PEP 8 says that the names declared in `__all__` are public APIs: https://pep8.org/#public-and-internal-interfaces

The API reference documentation shows these classes under `session_management` and hides the private submodules:
image

File: ni_measurementlink_service/session_management/_types.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+        # The protobuf PinMapContext sites field is a RepeatedScalarContainer, not a list.
+        # Constructing a protobuf PinMapContext with sites=None sets sites to an empty
+        # RepeatedScalarContainer, not None.
+        return PinMapContext(
+            pin_map_id=other.pin_map_id,
+            sites=list(other.sites),
+        )
+
+    def _to_grpc(self) -> pin_map_context_pb2.PinMapContext:
+        return pin_map_context_pb2.PinMapContext(pin_map_id=self.pin_map_id, sites=self.sites or [])
```
Comment (bkeryan) [inline ni_measurementlink_service/session_management/_types.py:55]: `or []` is unnecessary

**Comment by bkeryan** [comment]
FYI @pbirkhol-ni To more clearly show the changes, I rewrote the history to fork `session_management.py` into all of the submodules in one commit and remove the duplicated parts in a separate commit. (I should have done that in the first place.) I will rebase and merge instead of squash-merging.

**Pull request merged**